### ISSUE TITLE
[2.21-2.23] backup from standby fails: unable to find primary cluster

### ISSUE THREAD

**Issue by psuderevsky** [pgbackrest/pgbackrest#924]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: tested 2.21,2.23

2. PostgreSQL version: 11.6

3. Operating system/version - Centos 7.6

4. Did you install pgBackRest from source or from a package? - package

5. Config:
```
[platform]
pg1-path=/db/postgres/11/pgdata
repo1-cipher-pass=<redacted>
repo1-cipher-type=aes-256-cbc
repo1-s3-bucket=<redacted>
repo1-s3-endpoint=<redacted>
repo1-path=/bucket
repo1-s3-region=<redacted>
repo1-s3-key=<redacted>
repo1-s3-key-secret=<redacted>
repo1-type=s3<redacted>
repo1-s3-verify-ssl=n
repo1-retention-full=5
repo1-retention-diff=3
process-max=4
start-fast=n
stop-auto=n
backup-standby=n
compress=y
compress-level=5
spool-path=/dbwal/backrest
buffer-size=8388608

[platform:backup]
pg1-host=localhost
pg1-host-port=2293
pg1-path=/db/postgres/11/pgdata
pg2-path=/db/postgres/11/pgdata
compress-level=6
process-max=32

[platform:restore]
process-max=64 

[platform:archive-get]
archive-async=y
archive-get-queue-max=2GB
process-max=6

[platform:archive-push]
archive-async=y
archive-push-queue-max=8GB
process-max=6
```

7. Describe the issue:
Backup can not be taken from standby. Works on 2.19 and doesn't on 2.21, 2.23.
Command:
```
pgbackrest --type=incr --backup-standby  backup  
```
pgBackRest connects (should) to master via local haproxy 2293 port.
```
2020-01-28 11:21:38.166 P00   INFO: backup command begin 2.23: --backup-standby --buffer-size=8388608 --compress --compress-level=6 --log-level-console=debug --pg1-host=localhost --pg1-host-port=2293 --pg1-path=/db/postgres/11/pgdata --pg2-path=/db/postgres/11/pgdata --process-max=1 --repo1-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo1-path=<redacted> --repo1-retention-diff=3 --repo1-retention-full=5 --repo1-s3-bucket=<redacted> --repo1-s3-endpoint=<redacted> --repo1-s3-key=<redacted> --repo1-s3-key-secret=<redacted> --repo1-s3-region=<redacted> --no-repo1-s3-verify-tls --repo1-type=s3 --stanza=platform --no-start-fast --no-stop-auto --type=incr
2020-01-28 11:21:39.243 P00   WARN: unable to check pg-1: [LockAcquireError] raised from remote-0 protocol on 'localhost': raised from remote-0 protocol on 'localhost': unable to acquire lock on file '/tmp/pgbackrest/platform-backup.lock': Resource temporarily unavailable
                                    HINT: is another pgBackRest process running?
                                    protocol/client:protocolClientProcessError:179:(trace log level required for parameters)
                                    protocol/client:protocolClientReadOutput:(trace log level required for parameters)
                                    protocol/client:protocolClientExecute:(trace log level required for parameters)
                                    protocol/client:protocolClientNoOp:(trace log level required for parameters)
                                    protocol/client:protocolClientNew:(trace log level required for parameters)
                                    protocol/helper:protocolRemoteGet:(debug log level required for parameters)
                                    storage/remote/protocol:storageRemoteProtocol:(debug log level required for parameters)
                                    protocol/server:protocolServerProcess:(debug log level required for parameters)
                                    command/remote/remote:cmdRemote:(debug log level required for parameters)
                                    main:main:(debug log level required for parameters)
2020-01-28 11:21:39.748 P00  ERROR: [056]: unable to find primary cluster - cannot proceed
2020-01-28 11:21:40.149 P00   INFO: http statistics: objects 1, sessions 1, requests 18, retries 0, closes 0
2020-01-28 11:21:40.149 P00   INFO: backup command end: aborted with exception [056]
```

**Comment by dwsteele** [comment]
Hmmm - it looks the the remote is trying to start its own remote.  Not sure how that is happening.

Can you provide debug level output?

**Comment by psuderevsky** [comment]
Here you are.
[pgbackrest_924.log](https://github.com/pgbackrest/pgbackrest/files/4123956/pgbackrest_924.log)

**Comment by dwsteele** [comment]
It looks like this is related to 8d3710b2.  This was added to deal with archive-push/get-async commands not picking up options in the config file.

It seems you have the same config on your primary so now it is picking up the `pg1-host` setting and trying to connect to that.  Note the `remote-0` repetition in the error:
```
unable to check pg-1: [LockAcquireError] raised from remote-0 protocol on 'localhost': raised from remote-0 protocol on 'localhost': unable to acquire lock on file '/tmp/pgbackrest/platform-backup.lock': Resource temporarily unavailable
```
We'll look at that,but meanwhile if you remove `pg1-host` from the primary I think it will work again.  That will make your config files different on the primary and standby.  Are you using exactly the same configuration on the primary and standby?  If so, this only accidentally worked before since we don't test this configuration -- or recommend it, since we think the configs must be asymmetric.

**Comment by psuderevsky** [comment]
Yes, my idea was to make absolutely the same pgbackrest config on each node (and on 2.19 it works).
Each standby has a cron task to run bash script, if the node is async standby then backup starts. In such way it works automatically regardless of current node roles in the cluster and keeps working after automatic failover without need to adjust config files.

**Comment by dwsteele** [comment]
That's a good goal.  To be honest, I'm not sure why it is working at all.  The `pg1-host` setting on the primary should confuse the `archive-get` command.  Is there anything special about your archive command, i.e. did you add `--reset-pg1-host`?

**Comment by psuderevsky** [comment]
No, I just placed `pg1-host` setting in `[platform:backup]` while `[platform:archive-get]` doesn't inherit it, but only settings from `[platform]`.
Also I set `backup-standby=n` in `[platform]` to keep ability of taking backups from master if needed and pass `--backup-standby` option explicitly in backup script. 

Looks like I've accomplished my goal on **2.23** by replacing
```
[platform:backup]
pg1-host=localhost
pg1-host-port=2293
pg1-path=/db/postgres/11/pgdata
pg2-path=/db/postgres/11/pgdata
compress-level=6
process-max=32
```
with 
```
[platform:backup]
pg2-host=localhost
pg2-host-port=2293
pg1-path=/db/postgres/11/pgdata
pg2-path=/db/postgres/11/pgdata
compress-level=6
process-max=32
```

**Comment by dwsteele** [comment]
That looks like it would do the trick, but it's a little confusing.  The local cluster should always be in position 1 so in some ways this is better.

We'll still need to fix remotes picking up host configurations from the remote host, but looks like you are OK for now.

**Comment by dwsteele** [comment]
This should be fixed now, see #1050.
